tools/codegen/dest/gen_external_aten_fallbacks.py

```diff
@@ -194,24 +156,18 @@ def get_device_param(args: List[Argument]) -> str:
                 return f"  static {dispatcher_sig.decl()};"
 
             elif self.target is Target.REGISTRATION:
-                if self.backend_index.has_backend(f):
-                    # xla has their own kernel: register it
-                    namespace = 'AtenXlaType'
-                else:
-                    # xla doesn't have a kernel: register the cpu fallback (or codegen'd out kernel).
-                    namespace = 'AtenXlaTypeDefault'
-                payload = f"static_cast<{dispatcher_sig.ptr_type()}>(&{namespace}::{name})"
+                # This codegen is only responsible for registering CPU fallback kernels
+                # We also skip registrations if there is a functional backend kernel,
+                # because we generate out/inplace wrappers in that case (handled in register_dispatch_key.py).
+                if self.backend_index.get(f) is not None or \
+                        (isinstance(g, NativeFunctionsGroup) and self.backend_index.get(g.functional) is not None):
```

Comment:
The downside to having this split in a separate dest here is that you have to keep the logic here in sync with register_dispatch_key.py for when to generate out/inplace wrappers. This is probably... OK, because there are fairly large benefits to having this be a separate file, but it's probably good to explicitly cross ref the corresponding condition and/or have a single function where you do the test that commons up the logic (even though it is short to type).